src/systemdhook.c

```diff
@@ -691,7 +692,7 @@ static int prestart(const char *rootfs,
 			return -1;
 		}
 
-		rc = dprintf(fd, "%.32s\n", id);
+		rc = dprintf(fd, "%.32s\n", container_uuid);
```

Comment:
I'm not familiar with dprintf but will assume that if you get a string that's less that 32 char that it will be OK?  If container_uuid's length is less that 32, should we just error out near the top of this function?

Reply:
It will end up with a string < 32 chars, which will in turn result in systemd not accepting it as a valid machine-id, thus creating a random machine-id on boot of the container.